FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c

```diff
@@ -572,59 +556,431 @@ The kernel will be configured as follows:
-        vTaskDelete(xTaskHandles[i]);
-    }
+  /* Delete all priority task responsibly*/
+  for (i = 0; i < configNUMBER_OF_CORES; i++) {
+    vTaskDelete(xTaskHandles[i]);
+  }
 }
 
-
-
-
 /*
 The kernel will be configured as follows:
     #define configNUMBER_OF_CORES                               (N > 1)
     #define INCLUDE_xTaskDelayUntil                          1
 
-Coverage for: 
+Coverage for:
         BaseType_t xTaskDelayUntil( TickType_t * const pxPreviousWakeTime,
                                     const TickType_t xTimeIncrement )
 */
-void test_task_delay_until_with_config_assert( void )
-{
-    TaskHandle_t xTaskHandles[configNUMBER_OF_CORES] = { NULL };
-    uint32_t i;
-    TickType_t previousWakeTime = xTickCount - 3;
-
-    /* Create tasks of equal priority for all available CPU cores */
-    for (i = 0; i < configNUMBER_OF_CORES; i++) {
-        xTaskCreate( vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 2, &xTaskHandles[i] );
+void test_task_delay_until_with_config_assert(void) {
+  TaskHandle_t xTaskHandles[configNUMBER_OF_CORES] = {NULL};
+  uint32_t i;
+  TickType_t previousWakeTime = xTickCount - 3;
+
+  /* Create tasks of equal priority for all available CPU cores */
+  for (i = 0; i < configNUMBER_OF_CORES; i++) {
+    xTaskCreate(vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 2,
+                &xTaskHandles[i]);
+  }
+
+  vTaskStartScheduler();
+
+  xTaskDelayUntil(&previousWakeTime, 4);
+}
+
+/**
+ * @brief vTaskGetInfo - populate TaskStatus_t and eTaskState
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *        pxTCB = prvGetTCBFromHandle( xTask );
+ *          ...
+ * @endcode
+ *
+ * Cover the case where xTask is NULL, and the current task is implicitly
+ * referenced and returned by prvGetTCBFromHandle(...);
+ */
+void test_coverage_vTaskGetInfo_implicit_task(void) {
+  TCB_t xTaskTCBs[1U] = {NULL};
+  TaskStatus_t pxTaskStatus;
+  BaseType_t xFreeStackSpace = pdFALSE;
+  eTaskState taskState = eReady;
+
+  xTaskTCBs[0].uxPriority = 1;
+  xTaskTCBs[0].xTaskRunState = 0;
+  xYieldPendings[0] = pdFALSE;
+  pxCurrentTCBs[0] = &xTaskTCBs[0];
+
+  uxTopReadyPriority = 1;
+  uxSchedulerSuspended = pdTRUE;
+
+  vTaskGetInfo(NULL, &pxTaskStatus, xFreeStackSpace, taskState);
+  TEST_ASSERT_EQUAL((BaseType_t)1, pxTaskStatus.uxCurrentPriority);
+}
+
+/**
+ * @brief vTaskGetInfo - populate TaskStatus_t and eTaskState
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *        pxTCB = prvGetTCBFromHandle( xTask );
+ *          ...
+ * @endcode
+ *
+ * Cover the case where xTask is specified rather than NULL.
+ */
+void test_coverage_vTaskGetInfo_explicit_task(void) {
+  TCB_t xTaskTCBs[1U] = {NULL};
+  TaskStatus_t pxTaskStatus;
+  BaseType_t xFreeStackSpace = pdFALSE;
+  eTaskState taskState = eReady;
+
+  xTaskTCBs[0].uxPriority = 1;
+  xTaskTCBs[0].xTaskRunState = 0;
+  xYieldPendings[0] = pdFALSE;
+  pxCurrentTCBs[0] = &xTaskTCBs[0];
+
+  uxTopReadyPriority = 1;
+  uxSchedulerSuspended = pdTRUE;
```

Comment:
Should the scheduler be suspended for this particular coverage test? If so, then i would add that to the test description so its clear the code path the test should be taking. If not, i think you might be able to delete this line? (or set it to pdFalse)

Reply:
I think this comment applies to other tests below too